### PR TITLE
Integrate axe automated testing

### PR DESCRIPTION
**Pull request by naseberry**
#### What
Integrate axe automated testing to project.

#### Ticket
[Integrate automated accessibility testing](https://dsdmoj.atlassian.net/browse/CBO-772)
[Fix automated accessibility issue](https://dsdmoj.atlassian.net/browse/CBO-773)

#### Why
Add accessibility test coverage to our CI/CD workflow.
Note: automated tests will not identify all accessibility issue, but will help the team to identify some issues.

### CODE REVIEW

File: app/assets/stylesheets/_denote.scss [hunk 1]
```diff
@@ -3,9 +3,14 @@
 .form-input-denote {
   position: absolute;
   margin: 2px 0 0 2px;
-  padding: $gutter-one-sixth $gutter-one-third;
+  padding: $gutter-one-sixth + 1 $gutter-one-third + 2;
```
Comment (naseberry) [inline app/assets/stylesheets/_denote.scss]: UI fix for the currency field

File: app/assets/javascripts/modules/external_users/claims/CaseTypeCtrl.js [hunk 2]
```diff
@@ -39,7 +39,7 @@ moj.Modules.CaseTypeCtrl = {
   bindEvents: function () {
     var self = this;
 
-    $.subscribe('/onConfirm/claim_case_type_id-select/', function (e, data) {
+    $.subscribe('/onConfirm/case_type-select/', function (e, data) {
```
Comment (naseberry) [inline app/assets/javascripts/modules/external_users/claims/CaseTypeCtrl.js]: Fix show/hide of date fields depending on case type selection (Trial, Re-trial etc.)

File: app/assets/stylesheets/_denote.scss (hunk 1 above)
Comment (naseberry) [inline app/assets/stylesheets/_denote.scss]: UI fix for currency fields

File: app/assets/stylesheets/_shame.scss [hunk 3]
```diff
@@ -851,19 +851,6 @@ Styling for Claims Messaging
...
-        text-decoration: none;
-      }
-
-      a:hover {
-        text-decoration: underline;
-      }
-
-      color: $grey-1;
-    }
-
```
Comment (naseberry) [inline app/assets/stylesheets/_shame.scss]: Removal due to colour contrast issue, fallback to default hyperlink styles.

File: app/views/external_users/claims/basic_fees/_basic_fee_fields_primary.html.haml [hunk 4]
```diff
@@ -12,6 +12,7 @@
       = render partial: 'external_users/claims/basic_fees/offence_display', locals: { f: f }
 
     = f.hidden_field :fee_type_id, value: f.object.fee_type_id, class: 'js-fee-type'
+    = f.label :quantity, 'Fee calculator days', class: 'visually-hidden'
```
Comment (naseberry) [inline app/views/external_users/claims/basic_fees/_basic_fee_fields_primary.html.haml]: label association for input on line 16, input cannot be of type 'hidden'

File: features/page_objects/sections/expense_section.rb [hunk 5]
```diff
@@ -7,13 +7,12 @@ class ExpenseSection < SitePrism::Section
   #
   element :expense_type_dropdown, "#claim_expenses_attributes_0_expense_type_id"
   element :destination_label, ".fx-establishment-select label"
-  element :destination, "#claim_expenses_attributes_0_location"
-  element :distance, "#claim_expenses_attributes_0_distance"
-  element :quantity, "#claim_expenses_attributes_0_distance"
```
Comment (naseberry) [inline features/page_objects/sections/expense_section.rb]: Note: confirmed as dead code

File: features/claims/advocate/scheme_eleven/advocate_supplementary_claim_submit.feature [hunk 6]
```diff
@@ -23,22 +27,23 @@ Feature: Advocate tries to submit a supplementary claim for miscellaneous fees (
...
     And I should see a page title "Enter defendant details for advocate supplementary fee claim"
+    Then the page should be accessible within "#content"
     When I click "Continue" in the claim form
     Then I should be in the 'Miscellaneous fees' form page
     And I should see the advocate categories 'Junior,Leading junior,QC'
     And the following miscellaneous fee checkboxes should exist:
-      | section | fee_description |
-      | miscellaneous | Confiscation hearings (half day uplift) |
-      | miscellaneous | Confiscation hearings (half day) |
+      | section       | fee_description                          |
```
Comment (jsugarman) [inline features/claims/advocate/scheme_eleven/advocate_supplementary_claim_submit.feature]: **not a blocker**, one reason for not tabbing these out is that in some editors the its more likely to cause wrapping and difficulty reading. but just saying.

File: app/form_builders/adp_text_field.rb [hunk 7]
```diff
@@ -52,7 +52,7 @@ def has_errors?
   def to_html
     result = div_start
     # Issue: duplicate id, this breaks the label & input association
-    result += anchor
+    # result += anchor
```
Comment (jsugarman) [inline app/form_builders/adp_text_field.rb:55]: is this removal deliberate? does it not break error message linking?

File: Gemfile [hunk 8]
```diff
@@ -123,4 +123,5 @@ group :test do
   gem 'vcr'
   gem 'webmock'
   gem 'test-prof'
+  gem 'axe-matchers', '~> 2.5'
```
Comment (jsugarman) [inline Gemfile]: generally good practice to have gems listed alphabetically

**Review by jsugarman — APPROVED** [review]
👍 LGTM. just one question on the removal of the `anchor` code in the `adp_text_field`

File: app/form_builders/adp_text_field.rb (hunk 7 above)
Comment (naseberry) [reply app/form_builders/adp_text_field.rb:55]: This removal is deliberate as error message were not linked to the relevant input, instead it was linked to an anchor above the input form group. This was an accessibility issue.

File: features/claims/advocate/scheme_eleven/advocate_supplementary_claim_submit.feature (hunk 6 above)
Comment (naseberry) [reply features/claims/advocate/scheme_eleven/advocate_supplementary_claim_submit.feature]: Applied by either my beautifier or editorconfig, I will try to configure my editor to accommodate this.
